File: app/controllers/XMPP.java
package controllers;

import play.mvc.*;
import play.Logger;

public class XMPP extends Controller {
    public static void receiveMessage() {
        // Parse the message details
        String to = params.get("to");
        String from = params.get("from");
        String body = params.get("body");

        // LOG the request details
        Logger.debug("XMPP Message sent to " + to + " by " + from + " is " + body );
        ok();
    }
}

File: app/controllers/XMPP.scala
package controllers

import play.mvc.Controller
import utils.XMPPSend
import play.Logger

object XMPP extends Controller {
  def receiveMessage {
        // Parse the message details
        var to = params.get("to")
        var from = params.get("from")
        var body = params.get("body")

        // LOG the request details
        Logger.debug("XMPP Message sent to " + to + " by " + from + " is " + body )
        ok()
    }
}